### PR TITLE
Added new reference / Added Hardlink support / Performance considerations

### PR DESCRIPTION
**Pull request by Karl-WE**
- ReFS can now support Hardlinks under certain conditions
added ReFS version reference

### CODE REVIEW

**Comment by illfated** [comment]
About the link in the last line of the page:
- [ReFS Version history (unofficial)](https://gist.github.com/0xbadfca11/da0598e47dd643d933dc#file-windows-refs-versions-en-md) (https://gist.github.com/0xbadfca11/da0598e47dd643d933dc#file-windows-refs-versions-en-md)

Is that a temporary solution, until a docs.microsoft.com page is created containing the same information?

**Comment by eross-msft** [comment]
@jasongerend  Please review these changes and let  me know when this is ready to publish. Thanks!

**Comment by Karl-WE** [comment]
@illfated yes it is. So far there no changelog or matrix like this for ReFS officially.
Of course it we might think about such a paragraph and it would be helpful.

**Comment by PRMerger14** [comment]
@Karl-WE : Thanks for your contribution! The author(s) have been notified to review your proposed change. @gawatu

**Comment by illfated** [comment]
@Karl-WE : Will you be able to resolve the file conflict?

Page location & history sources:
- MS Docs: https://docs.microsoft.com/en-us/windows-server/storage/refs/refs-overview
- GitHub: https://github.com/MicrosoftDocs/windowsserverdocs/blob/master/WindowsServerDocs/storage/refs/refs-overview.md
- History: https://github.com/MicrosoftDocs/windowsserverdocs/commits/master/WindowsServerDocs/storage/refs/refs-overview.md

Looks like you have contributed quite a lot to this page yourself since this PR was created, do you know if any of your previous commits overlap the changed content in this PR?

**Comment by PRMerger10** [comment]
@Karl-WE : Thanks for your contribution! The author(s) have been notified to review your proposed change. @gawatu

**Comment by Karl-WE** [comment]
@illfated @gawatu the conflict has been resolved. I have also removed the mention of Windows 10 Pro for ReFS 3.5 as this version is not able to newly format ReFS drives. I hope the current version is now clear and ready to merge.

**Comment by PRMerger15** [comment]
@Karl-WE : Thanks for your contribution! The author(s) have been notified to review your proposed change. @gawatu

**Comment by Karl-WE** [comment]
@illfated  @gawatu this PR has a subsequent optional PR I did not merge it intentionally to give you time for review.
https://github.com/Karl-WE/windowsserverdocs/pull/1
I am not sure how I could mention / point you to this otherwise. Mention was not possible from there.

**Comment by illfated** [comment]
Fair enough, thank you for telling me about the PR upon which this PR relies. I will make a comment in the prioritized PR to keep track of it.

**Comment by PRMerger16** [comment]
@Karl-WE : Thanks for your contribution! The author(s) have been notified to review your proposed change. @gawatu

**Comment by PRMerger7** [comment]
@Karl-WE : Thanks for your contribution! The author(s) have been notified to review your proposed change. @gawatu

**Comment by Karl-WE** [comment]
now contains new chapter performance considerations that i previously handled in a private branch for discussion. as suggestion by @illfated I merged it with #4092 as it is unlikely the other branch would be read, due to time contraints from the docs team. 

for more information on the idea of adding this whole new chapter please read my notes here
https://github.com/MicrosoftDocs/windowsserverdocs/pull/4092/commits/13175bbaca89de05e9477850d8c35efc7d6d63d0

File: WindowsServerDocs/storage/refs/refs-overview.md
Comment (illfated) [inline WindowsServerDocs/storage/refs/refs-overview.md:184]: Hmm. I feel a bit uneasy with this phrasing and the use of the term "unlock" here. Looking at the how it currently reads, the meaning does not seem to be obvious, at least not to me. I would like to ask if you meant "break", "discontinue", "turn off", or "unlink", because the current phrasing looks like it means 'unlock' in the meaning of "unlock the next level" or "unlock a new feature" in the software. If you mean 'unlock' as in "turn off" or "break", I suggest using a term with a more direct and clear meaning, like for example:
```suggestion
Disclaimer: Any in-place upgrade to this version will upgrade ReFS attached volumes to the latest version, but will also break current hardlinks. 
```
My suggestion may very well be not precise enough, because I don't actually know exactly what happens to hardlinks when ReFS is upgraded from version 3.5 to a newer one.

File: WindowsServerDocs/storage/refs/refs-overview.md
Comment (illfated) [inline WindowsServerDocs/storage/refs/refs-overview.md:105]: The way I read the first sentence, is that it is meant to compare ReFS to NTFS and find ReFS to be better. In such comparisons, it is necessary to be more precise about the difference, like e.g. (for example):
```suggestion
ReFS is a superior filesystem to NTFS in many ways. It can greatly improve the performance in read / write operation scenarios. This is especially the case when comparing operations like creating Hyper-V virtual disks, snapshots or merging them. These performance benefits are not limited to Hyper-V though. 
```
Again, my suggestion may be imprecise, mostly due to my lack of knowledge about this software technology. I am mostly trying to convey that in a comparison between two systems, where one is better than the other (in one or more ways), the comparison needs to include a qualifier to say how one of them is better than the other. When using the preposition "over" in a comparison, you often need to rephrase the preceding text to say how the preference is made or how one of them takes precedence over the other.

File: WindowsServerDocs/storage/refs/refs-overview.md
Comment (illfated) [inline WindowsServerDocs/storage/refs/refs-overview.md:89]: Readability improvement suggestion:
```suggestion
> For CSV: NTFS should be used for traditional SANs. ReFS should be used on top of S2D. Refer to the functionality chapter.
```
A secondary phrasing could be: "Refer to the manual chapter concerning functionality."

File: WindowsServerDocs/storage/refs/refs-overview.md
Comment (illfated) [inline WindowsServerDocs/storage/refs/refs-overview.md:80]: Just like in my other comment, the phrase "Refer chapter functionality." uses incomplete English syntax. A better English phrasing would be:
```suggestion
> Please consider **limitations** on Cluster Shared Volumes in general, or CSVs used on Storage Spaces or Storage Spaces Direct (S2D). Refer to the functionality chapter.
```

File: WindowsServerDocs/storage/refs/refs-overview.md
Comment (illfated) [reply WindowsServerDocs/storage/refs/refs-overview.md:105]: Another way to phrase the first sentence could be:
```suggestion
ReFS is an advanced file system, superior to NTFS in many ways. It can greatly improve the performance in read / write operation scenarios. This is especially the case when comparing operations like creating Hyper-V virtual disks, snapshots or merging disks. These performance benefits are not limited to Hyper-V, though. 
```
I also threw in a suggestion for replacing "them", because "them" can be ambiguous here. Readability comma added near the end (natural speech). File System should most likely be separated and not in one word (in contrast to other germanic languages).

File: WindowsServerDocs/storage/refs/refs-overview.md
Comment (illfated) [inline WindowsServerDocs/storage/refs/refs-overview.md:110]:
```suggestion
While **supported**, scenarios exist where ReFS performance may not meet the expectations, compared to NTFS formatted volumes.
```

File: WindowsServerDocs/storage/refs/refs-overview.md
Comment (illfated) [inline WindowsServerDocs/storage/refs/refs-overview.md:114]:
```suggestion
Typical scenarios that may apply, based on the underlying hardware configuration, are typically Windows User Profiles, Temporary Folders of browsers in high-density or multi-user environments, such as Microsoft Remote Desktop Host (RDSH), Windows Virtual Desktop (WVD), or 3rd party applications like Citrix User Profile Management (UPM).
```

File: WindowsServerDocs/storage/refs/refs-overview.md
Comment (illfated) [reply WindowsServerDocs/storage/refs/refs-overview.md:184]: Depending on the correct understanding of the hardlink feature and upgrading ReFS to a newer version, it could also be phrased like this:
```suggestion
Disclaimer: Any in-place upgrade to this version will upgrade ReFS attached volumes to the latest version, but will disable existing usage of hardlinks. 
```

Comment (Karl-WE) [reply WindowsServerDocs/storage/refs/refs-overview.md:184]: Hi @illfated thanks for spotting this. It should be 

```suggestion
Disclaimer: Any in-place upgrade to this version will upgrade ReFS attached volumes to the latest ReFS version, but will **not** make the hardlinks feature available.
```

**Comment by PRMerger19** [comment]
@Karl-WE : Thanks for your contribution! The author(s) have been notified to review your proposed change. @gawatu

**Comment by PRMerger20** [comment]
@Karl-WE : Thanks for your contribution! The author(s) have been notified to review your proposed change. @gawatu

**Comment by PRMerger20** [comment]
@Karl-WE : Thanks for your contribution! The author(s) have been notified to review your proposed change. @gawatu

**Comment by PRMerger20** [comment]
@Karl-WE : Thanks for your contribution! The author(s) have been notified to review your proposed change. @gawatu

**Comment by PRMerger20** [comment]
@Karl-WE : Thanks for your contribution! The author(s) have been notified to review your proposed change. @gawatu

**Comment by eross-msft** [comment]
@Anhansen @nedpyle  Can either of your confirm whether this ready to publish? Or if not, point me to who can make the call. Thanks!

**Comment by PRMerger6** [comment]
@Karl-WE : Thanks for your contribution! The author(s) have been notified to review your proposed change. @gawatu

**Comment by PRMerger13** [comment]
@Karl-WE : Thanks for your contribution! The author(s) have been notified to review your proposed change. @gawatu

**Comment by Karl-WE** [comment]
@eross-msft with help from @illfated, I think from our end we have finished polishing the grammar and typos.

**Review by illfated — APPROVED** [review]
Looks good to me. Will defer to the document author for final review.

**Comment by sethmanheim** [comment]
Seems that none of these folks are around anymore, and the entire file is a merge conflict.

**Pull request closed without merging**
